Fix namespacing for multiple instances of gazebo_ros2_control plugin (#181)
ros-controls/gazebo_ros2_control: doc/index.rst

.. literalinclude:: ../gazebo_ros2_control_demos/config/cart_controller.yaml
   :language: yaml

Multiple Namespaces
-----------------------------------------------------------

> The gazebo_ros2_control plugin can be launched in multiple namespaces. This is useful if it is desired to have multiple robots of the same description running at the same time.

Reviewer: The gazebo_ros2-control plugin supports multiple namespaces

> The gazebo_ros2_control plugin can be launched in multiple namespaces. This is useful if it is desired to have multiple robots of the same description running at the same time.

Reviewer: Remove, the previous sentence is enough

> If the namespace is to be defined in the URDF file, then you can use the ``<namespace>`` tag as detailed below:

Reviewer: It is possible to directly define the namespace in the urdf file, as showed below:

.. code-block:: xml

  <gazebo>
    <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
      <parameters> $(arg control_yaml) </parameters>
      <ros>
        <namespace>r1</namespace>
        <remapping>/tf:=tf</remapping>
        <remapping>/tf_static:=tf_static</remapping>
        <remapping>diffdrive_controller/odom:=odom</remapping>
      </ros>
    </plugin>
  </gazebo>

> Where ``r1`` is the namespace that the plugin nodes will be run in. Please note that the remapping of ``/tf`` and ``/tf_static`` is recommended only if you want to have a separate TF buffer for each namespace, e.g ``/r1/tf``

Reviewer: associated to the particular instance of the plugin

> A second method for launching in namespaces is by using the Gazebo ``spawn_entity.py`` tool and setting the ``-robot_namespace`` parameter to the script:

Reviewer: A second method to launch an instance with a particular namespace is..

.. code-block:: bash

>   ros2 run gazebo_ros spawn_entity.py -entity robot_1 -robot_namespace r1 -topic robot_description

Reviewer: use r1 as above

.. literalinclude:: ../gazebo_ros2_control_demos/config/cart_controller.yaml
   :language: yaml

Multiple Namespaces
-----------------------------------------------------------
The gazebo_ros2_control plugin can be launched in multiple namespaces. This is useful if it is desired to have multiple robots of the same description running at the same time.

> If the namespace is to be defined in the URDF file, then you can use the ``<namespace>`` tag as detailed below:

Reviewer 2: ```suggestion
It is possible to directly define the namespace in the URDF file, as shown below:
```

.. code-block:: xml

  <gazebo>
    <plugin name="gazebo_ros2_control" filename="libgazebo_ros2_control.so">
      <parameters> $(arg control_yaml) </parameters>
      <ros>
        <namespace>r1</namespace>
        <remapping>/tf:=tf</remapping>
        <remapping>/tf_static:=tf_static</remapping>
        <remapping>diffdrive_controller/odom:=odom</remapping>
      </ros>
    </plugin>
  </gazebo>

> Where ``r1`` is the namespace that the plugin nodes will be run in. Please note that the remapping of ``/tf`` and ``/tf_static`` is recommended only if you want to have a separate TF buffer for each namespace, e.g ``/r1/tf``

Reviewer 2: ```suggestion
Where ``r1`` is the namespace associated to the particular instance of the plugin. Please note that the remapping of ``/tf`` and ``/tf_static`` is recommended only if you want to have a separate TF buffer for each namespace, e.g ``/r1/tf``
```

> A second method for launching in namespaces is by using the Gazebo ``spawn_entity.py`` tool and setting the ``-robot_namespace`` parameter to the script:

Reviewer 2: ```suggestion
A second method to launch an instance with a particular namespace is by using the Gazebo ``spawn_entity.py`` tool and setting the ``-robot_namespace`` parameter to the script:
```
